### PR TITLE
Feature/add request description

### PR DESCRIPTION
**Pull request by anhdat**
Paw's Request description is where we can add more information about the request like business rules, explain parameters of the request. Thus, it's an essential item when generating docs from Paw.

### CODE REVIEW

**Comment by kylef** [comment]
Thanks for the pull request @anhdat. It seems that when I don't supply a request description, the generated API Blueprint now has two extra new lines between the title and request/response example. 

Would you be able to restore previous behaviour here (single new line not triple) and I'll happy merge and release this in an update.

Before:

```apib
# GET /networks/1/

+ Request (text/plain; charset=utf-8)
```

After:

```apib
# GET /networks/1/



+ Request (text/plain; charset=utf-8)
```

Cheers!

**Comment by anhdat** [comment]
@kylef Thanks for the comment. 👍  I will update it right away.

File: apiblueprint.mustache [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # {{{ method }}} {{{ path }}}
 
+{{#description?}}
```
Comment (kylef) [inline apiblueprint.mustache]: Shouldn't this be `request.description?`? Now I'm not seeing descriptions.

Comment (anhdat) [inline apiblueprint.mustache]: Ah, my bad. Just updated the code to fix this.

**Comment by kylef** [comment]
Thanks @anhdat!

**Comment by anhdat** [comment]
Thanks @kylef . This would be the first for many PRs to come. I'm also working on support Name and Group for exporting with [aglio](https://github.com/danielgtaylor/aglio) to make generated file from extension ready to generate docs site. Though, I have some issues:

- With Name, Blueprint seems doesn't like single request, for now this extension produces `# {{{ method }}} {{{ path }}}` and aglio renders to this `POST/api/v1/auth/token/`, while with a request name like `Authenticate` would be more readable. 
- With Group, I will go with either two ways:
  - Only take the first item in Paw file, the fetch siblings and children to render the whole file
  - Mark each request with its parent in comment tags `<!-- -->`, then use a separate script to insert approriate Group marks

What do you think?

**Comment by kylef** [comment]
@anhdat I was actually thinking about refactoring this plugin so that it uses [Fury](https://github.com/apiaryio/fury.js) and our [Fury API Blueprint serialiser](https://github.com/apiaryio/fury-adapter-apib-serializer) instead. There is currently duplicate code between this plugin and the serialiser, and the serialiser already does a far better job (which isn't specific to this plugin). See #14.

This way, we would be constructing an ["API Element"](http://api-elements.readthedocs.io) from the Paw and the Fury plugin will serialise it. The API Blueprint serialiser already has better creation of an API Blueprint since it supports MSON structures etc.

---

If there is a blueprint name, you can break use the `# Name [METHOD PATH]` syntax.

**Comment by anhdat** [comment]
@kylef sounds interesting, I could dive into that this week. One question is #14 was opened from Sep '15, what was the obstacles that you meet when implement it then?

**Comment by kylef** [comment]
At the time, I suspected that fury and fury-adapters may not be capable of running inside Paw. But I think that is no longer the case and they should work. I don't think there are any obstacles, it's just been other priorities in the parser than the refactoring in this plugin hence why it hasn't been visited yet.

If you want to give it a go, that would be great. I'm happy to help if you hit any problems or have any question. I'm available on the api-blueprint slack (https://apiblueprint-slack.herokuapp.com/).

**Comment by pksunkara** [comment]
@anhdat I agree with @kylef. We can move with #14 since we are now sure that fury can work with PAW. I will be available if you have any questions.

**Pull request merged**